Exclude doc self

akatsoulas opened the pull request:
mozilla/sumo#1288

escattone reviewed (changes requested):
The fix for this turned out to be more extensive than I anticipated due to the caching that `_documents_for()` does. I decided to create the fix as a PR against this one:

https://github.com/akatsoulas/kitsune/pull/8

akatsoulas commented on kitsune/wiki/facets.py:
```diff
@@ -69,31 +73,36 @@ def documents_for(locale, topics=None, products=None):
     documents = cache.get(documents_cache_key)
+    exclude_current_document = isinstance(current_document, Document)
     if documents is not None:
+        if exclude_current_document:
```
I moved the exclusion here b/c of [this call](https://github.com/mozilla/kitsune/pull/5483/files#diff-754fad9f530ba68d2f0722f3572f6f0b45752ff296970f1839e01c660fbfd80fR63) to `_documents_for` . This way we are making sure that we are always excluding the document.

akatsoulas commented:
@escattone another pass?

akatsoulas commented on kitsune/wiki/facets.py:
```diff
@@ -69,31 +73,36 @@ def documents_for(locale, topics=None, products=None):
+        current_revision__isnull=False,
+        category__in=settings.IA_DEFAULT_CATEGORIES,
     )
+    if exclude_current_document:
+        qs = qs.exclude(id=current_document.id)
```
My main comment here and what I tried to do in the other PR is that we need to exclude the parent in all occasions, not just when the local is the same. Any reference and any relationship to the current_document should be excluded from the fallback_documents. I am going to clean up the PR following your pattern here and we can have a final pass

akatsoulas commented:
@escattone I squashed all of your work and mine in the second commit. I added a 3rd commit which is basically new code that you have not review. If this looks good I will squash the 3rd commit into the 2nd and this should be good to go. Final pass on that 3rd commit?

escattone reviewed (approved):
Looks great, thanks @akatsoulas!
